### PR TITLE
Remove componentWillReceiveProps usages in examples

### PR DESCRIPTION
**Pull request by chaitanya-deep**
**Summary**
Resolves [1940](https://github.com/facebook/draft-js/issues/1940)
Change usage of `componentWillReceiveProps` in Tex examples to `componentDidUpdate`

### CODE REVIEW

**Comment by claudiopro** [comment]
Hi @chaitanya-deep, welcome to Draft.js and thanks for taking the time to look into this issue! This change looks good to me 👍🏼, sounds good to remove the sketchy `setTimeout`s also.

File: examples/draft-0-9-1/tex/js/components/TeXBlock.js [hunk 1]
```diff
@@ -21,38 +21,26 @@ import {Entity} from 'draft-js';
...
+    );
   }
 
   componentDidMount() {
     this._update();
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.content !== this.props.content) {
+  componentDidUpdate(prevProps) {
```
Comment (claudiopro) [inline examples/draft-0-9-1/tex/js/components/TeXBlock.js]: Maybe worth calling out there's an unused `prevState` second argument here.

Comment (chaitanya-deep) [reply examples/draft-0-9-1/tex/js/components/TeXBlock.js]: Made the change but is that necessary?
The third `snapshot` argument is also unused.

And the docs for typical usage of componentDidUpdate don't include it either
https://reactjs.org/docs/react-component.html#componentdidupdate

File: package.json [hunk 2]
```diff
@@ -85,7 +85,7 @@
     "webpack-stream": "^4.0.0"
   },
   "resolutions": {
-    "gulp/**/natives": "1.1.3"
+    "gulp/**/natives": "1.1.6"
```
Comment (niveditc) [inline package.json]: @chaitanya-deep - can we split this part into a separate PR? Just so we don't conflate two different changes `componentWillReceiveProps` & dependency updates. 

Also for the dependency update, I think updating the `yarn.lock` file is a good idea.
